### ISSUE TITLE
[Multi viewport] Docking on central node issue

### ISSUE THREAD

**Issue by Vin789** [ocornut/imgui#6753]
**Version/Branch of Dear ImGui:**

Version: 1.89.9 WIP
Branch: docking

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_opengl3.h/cpp / imgui_impl_sdl2.h/cpp
Operating System: Windows 10

**My Issue/Question:**

Hi ! I have an issue with docking using multi viewport. I have my main window with a DockSpaceOverViewport. Then I create two others windows. Then when docking one into the other one using the central node the bug occurs (only with the central node).  It seems that the docking is not completed and the popup with the dock part stay on. Then switching from tabs fix the issue. It may be related to the fact of using some BeginChild/EndChild inside the sub windows.

**Screenshots/Video**

https://github.com/ocornut/imgui/assets/13332569/249fa4d9-2ecb-472c-b5e3-9bdbad9047ab

**Standalone, minimal, complete and verifiable example:** _(see https://github.com/ocornut/imgui/issues/2261)_
```
//Juste take the example_sdl2_opengl3 project. Inside main.cpp remove the code from point 1 to 3 (line 151 to 186) and use this one instead:

ImGui::DockSpaceOverViewport(nullptr, ImGuiDockNodeFlags_PassthruCentralNode);

if (ImGui::Begin("Asset Browser"))
{
    ImGui::BeginChild("Test child 1", ImVec2(200.0f, 0.0f), true, ImGuiWindowFlags_AlwaysAutoResize);
    ImGui::Text("Test text");
    ImGui::EndChild();

    ImGui::BeginChild("Test child 2", ImVec2(200.0f, 0.0f), true, ImGuiWindowFlags_AlwaysAutoResize);
    ImGui::Text("Test text");
    ImGui::EndChild();
}
ImGui::End();

if (ImGui::Begin("Property Grid 2"))
{
    ImGui::Text("CONTENT");
}
ImGui::End();


```

Any help is welcome. Thx for your time.

Vincent

**Comment by ocornut** [comment]
Thank you for your report and repro.
I can confirm this happens here and it also happens with other backends (I first though the rendering of that viewport was "frozen" but it's not). Investigating now.

**Comment by ocornut** [comment]
Pushed fix 762ec44

Clues:

(1) I noticed it could happen (e.g. at boot) without even doing a manual dock (based on existing docking state).

(2) Didn't happen without the Child windows. I thought it was related to child window decoration rendered in parent window's drawlist, but still happens with borders off.

(2) I changed some display from e.g.
```cpp
ImGui::Text("CONTENT");
```
to
```cpp
ImGui::Text(ImGui::GetIO().KeyShift ? "SHIFT" : "CONTENT");
```
To help understand what's happening. Framebuffer is not cleared when windows have a background, which is an optimization, but in this case **the background was not rendered which was causing our issue**.

(4) I recall I made changes recently in `Render()` to reorder some stuff to facilitate injection of ctrl+tab/modal dimming background. I updated to previous version and confirmed the bug started happening with f422e78 for #6716 (August 14).

Spot the bug:
```cpp

    for (int n = 0; n != g.WindowsFocusOrder.Size; n++)
    {
        ImGuiWindow* window = g.Windows[n];
        if (window->Flags & ImGuiWindowFlags_DockNodeHost)
            window->DrawList->ChannelsMerge();
    }
```
Spotted?

It never crashed because `g.Windows[]` is always larger than `g.WindowsFocusOrder[]`. The later only contains root windows, introducing child windows makes the arrays not match.

Thanks!
